state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -2772,8 +2772,14 @@ pub mod pallet {
 			)
 			.and_then(|amount| C::ChainAmount::try_from(amount).ok())
 			.unwrap_or_else(|| {
-				log::warn!("Unable to calculate input amount required for gas of {required_gas:?} for input asset ${input_asset:?}. Estimating the input amount based on a reference price.");
-				C::input_asset_amount_using_reference_gas_asset_price(input_asset,required_gas)
+				log::warn!("Unable to calculate input amount required for gas of {required_gas:?} for input asset ${input_asset:?}. Estimating the input amount based on either oracle price or reference price.");
+				let oracle_price = T::PriceFeedApi::get_price(input_asset.into());
+				let price = if oracle_price.as_ref().is_some_and(|price| !price.stale) {
+					Some(oracle_price.unwrap().price)
+				} else {
+					None
+				};
```

Comment:
We could just use the price even if it's stale? Even a stale price is likely to be better than the hard-coded estimate (which is by definition *more* stale)?